query_resolver_handlers.go

```diff
@@ -0,0 +1,336 @@
+package surrealdb
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/buger/jsonparser"
+	"github.com/goccy/go-json"
+)
+
+var (
+	// ErrResolvedQueryResultIsInvalid Need a better name :|
+	ErrResolvedQueryResultIsInvalid = errors.New("the result from the database response is not valid, expected an array")
```

Comment:
ErrInvalidRpcResult ? make it so that it covers more cases than just array expected, but just a general rpc result problem